cpp/search/search.h

```diff
@@ -18,6 +18,8 @@
 #include "../search/searchprint.h"
 #include "../search/timecontrols.h"
 
+#include "../external/nlohmann_json/json.hpp"
```

Comment:
This probably has the same problem as including this in rules, but it's not clear where else these functions belong. A recommendation for a home would be helpful.